src/core/ext/filters/client_channel/client_channel_factory.c

```diff
@@ -63,6 +63,6 @@ static const grpc_arg_pointer_vtable factory_arg_vtable = {
 
 grpc_arg grpc_client_channel_factory_create_channel_arg(
     grpc_client_channel_factory* factory) {
-  return grpc_channel_arg_pointer_create(GRPC_ARG_CLIENT_CHANNEL_FACTORY,
+  return grpc_channel_arg_pointer_create((char*)GRPC_ARG_CLIENT_CHANNEL_FACTORY,
```

Comment:
Instead of casting away the const, how about changing all of the `grpc_channel_arg_*_create()` functions to expect a const value as the parameter?